### ISSUE TITLE
TreeNodeArrays from children_nids and member_nids do not preserve tree ref

### ISSUE THREAD

**Issue by zack-vii** [MDSplus/mdsplus#292]
``` python
import MDSplus
tree = MDSplus.Tree('test',-1,'new')
tree.addNode('A','STRUCTURE')
tree.write()
tree.quit()
tree = MDSplus.Tree('test',-1)
node = tree.top
nodes = node.getChildren()
print(nodes[0],node)
# tree holds the ref to tree -> ctx is valid
del(tree)
# node.tree still holds the ref to tree -> ctx is still valid
print(nodes[0],node)
del(node)
# nodes.tree only holds the pointer information in ctx -> ctx is invalid
print(nodes[0])
```

the del(tree) is similar to exiting a function returning 'nodes'. outside of the function the return value is broken.

**Comment by zack-vii** [comment]
expected output:

```
(.A, \TEST::TOP)
(.A, \TEST::TOP)
.A
```

current output:

```
(.A, \TEST::TOP)
(.A, \TEST::TOP)
None
```

**Comment by zack-vii** [comment]
fixed by #294
